segfast/utils.py

```diff
@@ -35,6 +37,63 @@ def update(self, n=1):
 notifier = Notifier
 
 
+class TraceHeaderSpec:
+    """ Trace header class to store its name and byte position. By default, byte position is defined by name
+    accordingly to SEG-Y specification.
+
+    Parameters
+    ----------
+    name : str or int
+        If str, name of the header. If int, is interpreted as 'byte' and name will be default from spec.
+    byte : int, optional
+        Byte position of the header, by default None. If None, default byte position from the spec will be used.
+    dtype : int ot str, optional
+        dtype for header (e.g. 'i2', 'f4') or its length in bytes (then is interpreted as integer type).
+    """
+    TRACE_HEADER_SIZE = 240
+
+    STANDARD_BYTE_TO_HEADER = {v: k for k, v in segyio.tracefield.keys.items()}
+    START_BYTES = sorted(segyio.tracefield.keys.values())
+    STANDARD_START_BYTE_TO_LEN = {start: end - start
+                            for start, end in zip(START_BYTES, START_BYTES[1:] + [TRACE_HEADER_SIZE + 1])}
+
+    def __init__(self, name=None, start_byte=None, dtype=None, byteorder=None):
+        self.name = name or self.STANDARD_BYTE_TO_HEADER[start_byte]
+        self.start_byte = start_byte or segyio.tracefield.keys[name]
```

Comment:
Just for unification: you can define `STANDARD_HEADER_TO_BYTE` to match `STANDARD_BYTE_TO_HEADER` and use it here.